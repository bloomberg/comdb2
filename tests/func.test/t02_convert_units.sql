select '== test for convert_units ==' as t;
select convert_units('hi', 2, 3)
select convert_units(1, 2, 3)
select convert_units(1, 'hi', 3)
select convert_units(1, 'hi', 'hi')
select convert_units(1, null, 'bytes')
select convert_units(1000, null, 'bytes') as bytes
select convert_units(2000, null, 'bytes') as bytes
select convert_units(10000, null, 'bytes') as bytes
select convert_units(10*1024*1024, null, 'bytes') as bytes
select convert_units(cast(10*1024*1024*1024 as real), null, 'bytes') as bytes
select convert_units(cast(10*1024*1024*1024*1024 as real), null, 'bytes') as bytes
select convert_units(cast(10*1024*1024*1024*1024*1024 as real), null, 'bytes') as bytes
