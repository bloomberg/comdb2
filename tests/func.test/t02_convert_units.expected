(t='== test for convert_units ==')
[select convert_units('hi', 2, 3)] failed with rc 300 Expected numeric type as first argument to convert_units
[select convert_units(1, 2, 3)] failed with rc 300 Expected string type as second argument to convert_units
[select convert_units(1, 'hi', 3)] failed with rc 300 Expected string type as second argument to convert_units
[select convert_units(1, 'hi', 'hi')] failed with rc 300 Expected string type as second argument to convert_units
(convert_units(1, null, 'bytes')='1.000000')
(bytes='1000.000000')
(bytes='1.953125 KiB')
(bytes='9.765625 KiB')
(bytes='10.000000 MiB')
(bytes='10.000000 GiB')
(bytes='10.000000 TiB')
(bytes='10.000000 PiB')
